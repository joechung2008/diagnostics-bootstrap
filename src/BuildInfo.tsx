import { Table } from "react-bootstrap";

const BuildInfo: React.FC<BuildInfoProps> = ({ buildVersion }) => {
  const items = [
    {
      name: "Build Version",
      value: buildVersion,
    },
  ];

  return (
    <Table>
      <thead>
        <tr>
          <th>Name</th>
          <th>Value</th>
        </tr>
      </thead>
      <tbody>
        {items.map((item, idx) => (
          <tr key={idx}>
            <td>{item.name}</td>
            <td>{item.value}</td>
          </tr>
        ))}
      </tbody>
    </Table>
  );
};

export default BuildInfo;
