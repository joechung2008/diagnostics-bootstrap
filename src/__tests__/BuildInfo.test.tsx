import { render } from "@testing-library/react";
import { describe, it, expect } from "vitest";
import BuildInfo from "../BuildInfo";

describe("BuildInfo", () => {
  it("renders correctly", () => {
    const { container } = render(<BuildInfo buildVersion="1.0.0" />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
