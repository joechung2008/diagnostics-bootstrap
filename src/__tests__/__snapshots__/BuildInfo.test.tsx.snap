// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BuildInfo > renders correctly 1`] = `
<table
  class="table"
>
  <thead>
    <tr>
      <th>
        Name
      </th>
      <th>
        Value
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        Build Version
      </td>
      <td>
        1.0.0
      </td>
    </tr>
  </tbody>
</table>
`;
